modules/account-lib/src/coin/cspr/transferBuilder.ts

```diff
@@ -1,30 +1,46 @@
 import { BaseCoin as CoinConfig } from '@bitgo/statics/dist/src/base';
-import { BuildTransactionError, NotImplementedError, SigningError } from '../baseCoin/errors';
+import { PublicKey } from 'casper-client-sdk';
+import { BuildTransactionError, InvalidParameterValueError, SigningError } from '../baseCoin/errors';
 import { BaseKey } from '../baseCoin/iface';
+import { TransactionType } from '../baseCoin';
 import { TransactionBuilder, DEFAULT_M } from './transactionBuilder';
 import { Transaction } from './transaction';
+import { SECP256K1_PREFIX } from './constants';
+import { isValidPublicKey, isValidAmount, isValidTransferId } from './utils';
 
 export class TransferBuilder extends TransactionBuilder {
   private _toAddress: string;
   private _amount: string;
+  private _transferId: number;
 
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
   }
 
   /** @inheritdoc */
   protected async buildImplementation(): Promise<Transaction> {
-    throw new NotImplementedError('buildImplementation not implemented');
+    this._session = {
+      amount: this._amount,
+      target: PublicKey.fromHex(SECP256K1_PREFIX + this._toAddress),
+      id: this._transferId,
+    };
+    this.transaction.setTransactionType(TransactionType.Send);
+    return await super.buildImplementation();
   }
 
   /** @inheritdoc */
   initBuilder(tx: Transaction): void {
-    throw new NotImplementedError('initBuilder not implemented');
+    super.initBuilder(tx);
+    this.transaction.setTransactionType(TransactionType.Send);
+    // TODO(STLX-793): init to and amount

```

Comment:
isn't this available now with casper-sdk v1.0.12?

Reply:
Yes, but it will be done as part of STLX-793.